### PR TITLE
Added Percolation folder

### PR DESCRIPTION
**Pull request by awalker20**

### CODE REVIEW

File: Java/Percolation/Percolation.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+ * 
+ * @author Adam Walker
+ *
+ * Percolation is used to find under what conditions a liquid or gas will drain
+ * thorugh a porous material. An example when this would be used is in a system 
+ * with insulated and metallic materials. Percolation would be used to find the 
+ * percentage of metallic material needed to be an electrical conductor
+ *
+ */
+
```
Comment (manisaiprasad) [inline Java/Percolation/Percolation.java:14]: Add Time complexity

File: Java/Percolation/WeightedQuickUnionUF.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+
+
+//import edu.princeton.cs.algs4.StdIn;
+//import edu.princeton.cs.algs4.StdOut;
+
```
Comment (manisaiprasad) [inline Java/Percolation/WeightedQuickUnionUF.java:7]: Add Time complexity

File: Java/Percolation/Percolation.java (hunk 1 above)
Comment (awalker20) [reply Java/Percolation/Percolation.java:14]: Done

File: Java/Percolation/WeightedQuickUnionUF.java (hunk 2 above)
Comment (awalker20) [reply Java/Percolation/WeightedQuickUnionUF.java:7]: Done

**Pull request merged**